receiver/oracledbreceiver/scraper.go

```diff
@@ -828,3 +828,43 @@ func (s *oracleScraper) shutdown(_ context.Context) error {
 	}
 	return s.db.Close()
 }
+
+func (s *oracleScraper) setupResourceBuilder(rb *metadata.ResourceBuilder) *metadata.ResourceBuilder {
+	rb.SetOracledbInstanceName(s.instanceName)
+	rb.SetHostName(s.hostName)
+	rb.SetServiceInstanceID(s.serviceInstanceID)
+	return rb
+}
+
+func getInstanceID(instanceString string, logger *zap.Logger) string {
+	parts := strings.SplitN(instanceString, "/", 2)
+	hostAndPort := parts[0]
+	service := ""
+	if len(parts) == 2 {
+		service = "/" + parts[1]
+	}
+
+	host, port, err := net.SplitHostPort(hostAndPort)
+	if err != nil {
+		return handleInstanceIDErr(err, logger, port, service)
+	}
+
+	if strings.EqualFold(host, "localhost") || net.ParseIP(host).IsLoopback() {
+		host, err = os.Hostname()
+	}
+	if err != nil {
```

Comment:
I think this `if` should be next to the statement that potentially sets `err`. It's harder to follow when they're separated.